db.js

```diff
@@ -0,0 +1,24 @@
+const mongoose = require('mongoose');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+
+// A singleton to ensure we only start the database once
+// assign the MongoMemoryServer instance to mongoServer
+let mongoServer;
+const uri = 'mongodb+srv://Surya:sun@asapcluster.3osvgms.mongodb.net/?retryWrites=true&w=majority'
+
+
+const startDatabase = async () => {
```

Comment:
Add error handling to the `startDatabase` and `stopDatabase` functions. This will help to catch and handle any issues that might occur during the connection or disconnection process. [important]